test/jdk/java/nio/channels/Channels/WriteFullyMemorySize.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8316156
+ * @summary Ensure Channels.newOutputStream.write doe not overrun max memory
+ * @run main/othervm -XX:MaxDirectMemorySize=5M WriteFullyMemorySize
+ */
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
+import java.util.Random;
+
+public class WriteFullyMemorySize {
+    // this value must exceed MaxDirectMemorySize
+    private static final int SIZE = 10*1024*1024;
+
+    public static void main(String[] args) throws IOException {
+        byte[] b = new byte[SIZE];
+        Random rnd = new Random(System.nanoTime());
+        rnd.nextBytes(b);
+        ByteArrayInputStream bais = new ByteArrayInputStream(b);
+        Path target = Files.createTempFile("sna", "fu");
+        try {
+            Files.copy(bais, target, StandardCopyOption.REPLACE_EXISTING);
+            byte[] res = Files.readAllBytes(target);
+            if (!Arrays.equals(b, res))
+                throw new RuntimeException("Arrays are not equal");
```

Comment:
Will this 'throw new RuntimeException("Arrays are not equal");' be ever thrown ?.

I tested without fix and 'java.lang.OutOfMemoryError' wiil be thrown at line 49(Files.copy) so even in failing case 'RuntimeException' will not be thrown.

By looking the test name 'WriteFullyMemorySize'  it is not clear to me what it is trying to test. If  you wants to test 'Files.readAllBytes' as well then i will suggest you to  change the name of test or put some comment. It will help the future maintainer of the code.

Reply:
The `Arrays.equals` is anticipatory. The test is changed in a9af89cff869c441c6dd04f635a4d93e7757f2fa.